sdk/src/main/scala/com/horizen/storage/SidechainStateStorage.scala

```diff
@@ -161,14 +163,16 @@ class SidechainStateStorage(storage: Storage, sidechainBoxesCompanion: Sidechain
     }
   }
 
+  def getLastWithdrawalCertificatePreviousMcBlockHashOpt: Option[Array[Byte]] = storage.get(lastWithdrawalCertificatePreviousMcBlockHashKey).asScala.map(_.data)
```

Comment:
I think, that for last certificate it should be not "previous". but "end mc block hash". It's previous for current certificate, but nor for last one.
`getLastWithdrawalCertificatePreviousMcBlockHashOpt` -> `getLastCertificateEndEpochMcBlockHashOpt`
same for `lastWithdrawalCertificatePreviousMcBlockHashKey` -> `lastCertificateEndEpochMcBlockHashKey`

Reply:
agree